huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api_publishment.go

```diff
@@ -0,0 +1,245 @@
+			},
+			"roma_app_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: `The application ID for ROMA APIC.`,
+			},
+
+			// Attributes
+			"publish_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The publish ID.`,
+			},
+		},
+	}
+}
+
+func resourceDataServiceApiPublishmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg    = meta.(*config.Config)
+		region = cfg.GetRegion(d)
+	)
+	client, err := cfg.NewServiceClient("dataarts", region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio client: %s", err)
+	}
+
+	err = publishApi(client, d)
+	if err != nil {
+		return diag.Errorf("error publishing API: %s", err)
+	}
+
+	resourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(resourceId)
+
+	return resourceDataServiceApiPublishmentRead(ctx, d, meta)
+}
+
+func queryApiPublishInfo(client *golangsdk.ServiceClient, workspaceId, apiId string) ([]interface{}, error) {
+	var (
+		httpUrl = "v1/{project_id}/service/apis/{api_id}/publish-info"
+	)
+
+	queryPath := client.Endpoint + httpUrl
+	queryPath = strings.ReplaceAll(queryPath, "{project_id}", client.ProjectID)
+	queryPath = strings.ReplaceAll(queryPath, "{api_id}", apiId)
+
+	opt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json;charset=utf8",
+			"workspace":    workspaceId,
+			"Dlm-Type":     "EXCLUSIVE",
+		},
+	}
+
+	requestResp, err := client.Request("GET", queryPath, &opt)
+	if err != nil {
+		return nil, common.ConvertExpected400ErrInto404Err(err, "error_code", exclusiveApiNotPublishErrors...)
+	}
+	respBody, err := utils.FlattenResponse(requestResp)
+	if err != nil {
+		return nil, err
+	}
+	return utils.PathSearch("publish_messages", respBody, make([]interface{}, 0)).([]interface{}), nil
+}
+
+// QueryApiPublishInfoByInstanceId is a method that used to query the publish record in the specified cluster ID.
+// Returns a 404 error if no publish record found.
+func QueryApiPublishInfoByInstanceId(client *golangsdk.ServiceClient, workspaceId, apiId,
+	instanceId string) (interface{}, error) {
+	publishRecords, err := queryApiPublishInfo(client, workspaceId, apiId)
+	if err != nil {
+		return nil, err
+	}
+	publishRecord := utils.PathSearch(fmt.Sprintf("[?instance_id=='%s']|[0]", instanceId), publishRecords, nil)
+	// Since this resource does not provide the import function, the release operation must have been performed before entering the query stage.
+	// So, the 'API_STATUS_CREATED' status does not exist.
+	// If the status response is empty (means not found) or value 'API_STATUS_OFFLINE', that means the API has been unpublished.
+	apiStatus := utils.PathSearch("api_status", publishRecord, "").(string)
```

Comment:
```suggestion
	apiStatus := utils.PathSearch(fmt.Sprintf("[?instance_id=='%s']|[0].api_status", instanceId), publishRecord, "").(string)
```

Reply:
Negative, the API record has been already filter by instance ID and returns an object for publish info.
It is unnecessary to filter again and this JMES expression is incorrect.